tools/codegen/dest/lazy_ir.py

```diff
@@ -58,6 +73,37 @@ def aten_symbol(schema: LazyIrSchema) -> str:
         return f'c10::Symbol::fromQualString("aten::{schema.aten_name}")'
     return f'at::aten::{schema.aten_name}'
 
+def cached_shape_inference(func: NativeFunction, schema: LazyIrSchema,
+                           returns_length: int, all_types: List[NamedCType]) -> str:
+    # call the meta kernel if it exists, to compute output shape/dtype for our IR
+    if func.structured or func.structured_delegate is not None:
```

Comment:
we are effectively caching on a rank here?

Reply:
we are caching on dag-hash, which currently includes shape but after your change would only include rank.  but more to the point, we're caching on the same thing we use for lazy graph cache, so it should make this shape inference caching always consistent with non-cached verison, right?